api/ruxos_posix_api/src/imp/fs.rs

```diff
@@ -402,3 +460,84 @@ pub fn sys_fchownat(
     );
     syscall_body!(sys_fchownat, Ok(0))
 }
+
+/// read value of a symbolic link relative to directory file descriptor
+///
+///
+pub fn sys_readlinkat(
+    fd: c_int,
+    pathname: *const c_char,
+    buf: *mut c_char,
+    bufsize: usize,
+) -> usize {
+    let path = char_ptr_to_str(pathname);
+    debug!(
+        "sys_readlinkat <= path = {:?}, fd = {:}, bufsize = {:}",
+        path, fd, bufsize
+    );
+    syscall_body!(sys_readlinkat, {
+        let mut options = OpenOptions::new();
+        options.read(true);
+        let dst = unsafe { core::slice::from_raw_parts_mut(buf as *mut u8, bufsize as _) };
+        // if fd == AT_FDCWD then readat the relative path
+        if fd == ctypes::AT_FDCWD as c_int {
+            let file = ruxfs::fops::File::open(path?, &options)?;
+            let file = File::new(file);
+            Ok(file.read(dst)?)
+        } else {
+            let dir = Directory::from_fd(fd)?;
+            let mut file = dir.inner.lock().open_file_at(path?, &options)?;
+            Ok(file.read(dst)?)
+        }
+    })
+}
+
+type LinuxDirent64 = ctypes::dirent;
+
+fn convert_name_to_array(name: &[u8]) -> [i8; 256] {
```

Comment:
I think there can be a way to remove this function, like `try_into()`. Slice and array can use `into/from` to convert. Currently it is acceptable.